Fixed being able to identify falsewalls by right clicking them

DeityLink opened the pull request:
Fixes #30512

:cl:
* bugfix: Fixed being able to identify falsewalls by right clicking them.
* bugfix: Fixed false reinforced walls not having the same description as reinforced walls.
* rscadd: False walls now give an extra message on examination when standing adjacent to them.

Exxion commented:
OK what about mesons just seeing void

DeityLink commented:
> 
> 
> OK what about mesons just seeing void

They will only see void if the falsewalls are open. They show up on mesons.

DeityLink commented:
Actually they should show on mesons now that you mention it since otherwise you can already identify falsewalls this way 
![image](https://user-images.githubusercontent.com/7573912/132129660-435e72c3-3267-47cc-9b50-009af48b070f.png)

Exxion commented:
You can see the floor with mesons even if it has `invisibility = 101`? That's very odd.

DeityLink commented:
> 
> 
> You can see the floor with mesons even if it has `invisibility = 101`? That's very odd.

nah, that's another problem I'm bringing up. They'll just see void.

DeityLink commented:
This doesn't remove the typo when examining them but I think that one is at least a bit more thematic, compared to the completely client-side act of right clicking.

DeityLink commented:
By @DamianX 's request I've also fixed the typos.

I changed the fix so it removes mouse_opacity instead of making invisibility = 101, so you will not see a black void while wearing mesons (though you can still use mesons and notice the wrong wall smoothing, and that you can't right click a floor)

Exxion commented:
>I've also fixed the typos.
Boo

boy2mantwicethefam commented:
Make false doors show up on mesons, I'm tired of the maint hideout discovery memes

DeityLink commented:
> 
> 
> Make false doors show up on mesons, I'm tired of the maint hideout discovery memes

Tell you what I'm gonna try later. I think I see how I could do that.

Kurfursten commented:
I thought the different wall description was a feature. Just look at it closely.

DeityLink commented:
> 
> 
> I thought the different wall description was a feature. Just look at it closely.

I thought as well but then we noticed the typo was also in the regular r_wall description. So we deduced that the typo was originally everywhere, and only fixed properly in regular walls.

DeityLink commented:
> 
> 
> Make false doors show up on mesons, I'm tired of the maint hideout discovery memes

Victory


https://user-images.githubusercontent.com/7573912/132132978-57be9c60-2010-4e2f-81c5-44992d299379.mp4

BomberBro commented:
I think the typo is fine, since when were false walls supposed to be perfect facsimiles of real walls?

DeityLink commented:
I'd rather that instead of a typo you get an extra message on examine() when adjacent

DamianX commented on code/modules/clothing/glasses/scanners.dm:
```diff
@@ -158,6 +161,41 @@ var/list/meson_wearers = list()
+		apply()
+		return
+
+	if (new_mob != viewing)
```
isn't this always true?

DeityLink replied:
Nope. The proc is called with a null arg to clear() and remove the viewer. Same as with the material scanner below.

DamianX replied:
The `if` before this one checks whether `new_mob` is equal to `viewing`. If it is, it returns. So the only way to reach this second `if`, is for `new_mob` to not be equal to `viewing`.

DeityLink replied:
oh, well then I fix it in both places

BomberBro commented:
> I'd rather that instead of a typo you get an extra message on examine() when adjacent

That sounds good too, if false walls had any real effort put into them I'd be all for making them extremely sneaky, but as it stands they take about 2 seconds to set up

DeityLink commented:
> 
> 
> > I'd rather that instead of a typo you get an extra message on examine() when adjacent
> 
> That sounds good too, if false walls had any real effort put into them I'd be all for making them extremely sneaky, but as it stands they take about 2 seconds to set up

To be fair as long as a light push is all it takes top open them, they really shouldn't take much more effort.

TGFH commented:
Why shouldn't false walls be identifiable with mesons? It would be stupid if literally the only way to identify them is to spam click walls.

Exxion commented:
Yeah this PR is really sprawling away from its original purpose

SonixApache commented:
Maybe split off the meson thing to its own PR? I for one would like it, it'd help maint hiders actually hide in maint.

Exxion commented:
Meson thing should definitely be its own PR. The rest of it is probably related enough to stay as one PR even if I'd rather it were two.

DeityLink commented:
> 
> People get very upset about random wallclicking because it's basically just metagaming.

Yeah my point is that metagamers shouldn't be able to hide so easily obvious attempts at metagaming.

DeityLink commented:
Meson changes split off to #30598
